plugins/locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayer.java

```diff
@@ -0,0 +1,167 @@
+package com.mapbox.mapboxsdk.plugins.locationlayer;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Bitmap;
+import android.graphics.drawable.Drawable;
+import android.support.annotation.ColorInt;
+import android.support.annotation.FloatRange;
+import android.support.annotation.Nullable;
+import android.support.annotation.StyleRes;
+import android.support.v4.content.ContextCompat;
+
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.style.layers.FillLayer;
+import com.mapbox.mapboxsdk.style.layers.Layer;
+import com.mapbox.mapboxsdk.style.layers.Property;
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory;
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+import com.mapbox.services.commons.geojson.Feature;
+import com.mapbox.services.commons.geojson.FeatureCollection;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.mapbox.mapboxsdk.plugins.locationlayer.Utils.getBitmapFromDrawable;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.visibility;
+
+final class LocationLayer {
+
+  private List<String> layerIds;
+
+  private MapboxMap mapboxMap;
+  private Context context;
+  private int styleRes;
+
+  LocationLayer(MapView mapView, MapboxMap mapboxMap, @StyleRes int styleRes) {
+    this.mapboxMap = mapboxMap;
+    context = mapView.getContext();
+    this.styleRes = styleRes;
+    initialize();
+  }
+
+  private void initialize() {
+    layerIds = new ArrayList<>();
+
+    addSources();
+
+    TypedArray typedArray = context.obtainStyledAttributes(styleRes, R.styleable.LocationLayer);
+
+    int drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_foregroundDrawable, -1);
+    addForegroundLayer(ContextCompat.getDrawable(context, drawableResId));
+    drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_backgroundDrawable, -1);
+    addBackgroundLayer(ContextCompat.getDrawable(context, drawableResId));
+    drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_bearingDrawable, -1);
+    addBearingLayer(ContextCompat.getDrawable(context, drawableResId));
+    drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_navigationDrawable, -1);
+    addNavigationLayer(ContextCompat.getDrawable(context, drawableResId));
+
+    @FloatRange(from = 0, to = 1.0)
```

Comment:
Took the annotations out and now throw a runtime exception if the value was given is outside 0 and 1.